GEOSwiftMapboxGL/MapboxGLAnnotations.swift

```diff
@@ -17,112 +24,93 @@ public protocol GEOSwiftMapboxGL {
-    // Cannot invoke initializer for type ... with an argument list of type (geometryCollection: GeometryCollection<T>)
-    // 1. Expected an argument list of type (geometryCollection: GeometryCollection<T>)
-    required public init<GEOSwiftMapboxGL>(geometryCollection: GeometryCollection<GEOSwiftMapboxGL>) {
-        let shapes = geometryCollection.geometries.map({ (geometry: GEOSwiftMapboxGL) ->
-            MGLShape in
-            return geometry.mapboxShape()
-        })
+extension MGLPolygon {
+  convenience init(linearRing: Polygon.LinearRing) {
+    var coordinates = linearRing.points.map(CLLocationCoordinate2D.init)
+    self.init(coordinates: &coordinates, count: UInt(coordinates.count))
+  }
+  
+  convenience init(polygon: Polygon) {
+    var exteriorCoordinates = polygon.exterior.points.map(CLLocationCoordinate2D.init)
+    self.init(
+      coordinates: &exteriorCoordinates,
+      count: UInt(exteriorCoordinates.count),
+      interiorPolygons: polygon.holes.map(MGLPolygon.init)
+    )
+  }
+}
 
-        if let coordinate = geometryCollection.centroid()?.coordinate {
-            self.centroid = CLLocationCoordinate2DFromCoordinate(coordinate)
-        } else {
-            self.centroid = CLLocationCoordinate2DFromCoordinate(CoordinateFromCLLocationCoordinate2D(CLLocationCoordinate2DMake(0, 0)))
-        }
 
-        self.shapes = shapes
-        
-        if let envelope = geometryCollection.envelope() as? Polygon {
-            let exteriorRing = envelope.exteriorRing
-            let sw = CLLocationCoordinate2DFromCoordinate(exteriorRing.points[0])
-            let ne = CLLocationCoordinate2DFromCoordinate(exteriorRing.points[2])
-            self.overlayBounds = MGLCoordinateBounds(sw:sw, ne:ne)
-        } else {
-            let zeroCoord = CLLocationCoordinate2DMake(0, 0)
-            self.overlayBounds = MGLCoordinateBounds(sw:zeroCoord, ne:zeroCoord)
-        }
 
-        super.init()
-    }
-    
-    required public init?(coder aDecoder: NSCoder) {
-        fatalError("init(coder:) has not been implemented")
-    }
-    
-    override public var coordinate: CLLocationCoordinate2D { get {
-        return centroid
-        }}
-    
-    // TODO: implement using "intersect" method (actually it seems that mapboxgl never calls it...)
-    public func intersects(_ overlayBounds: MGLCoordinateBounds) -> Bool {
-        return true
-    }
+extension Geometry : GEOSwiftMapboxGL {
+    public func mapboxShape() -> MGLShape {
+        switch self {
+            
+        case let .point(self):
+            return MGLPointAnnotation(point: self)
+            
+        case let .lineString(self):
+            return MGLPolyline(lineString: self)
+
+        case let .polygon(self):
+            return MGLPolygon(polygon: self)
+
+        case let .multiPolygon(self):
+            return MGLMultiPolygon(polygons: self.polygons.map(MGLPolygon.init))
```

Comment:
Could we split this out into an extension on `MGLMultiPolygon` instead so that this case could be just like the ones above that simply construct a single Mapbox object from a single GEOSwift value? (same goes for the 3 cases that follow)